Casks/b/baiduinput.rb

```diff
@@ -1,15 +1,18 @@
 cask "baiduinput" do
-  version "1.0"
-  sha256 :no_check
+  version "6.0.3.66,1000e"
+  sha256 "4d31a7a0dbf09d32eaecc1f0b778f27128707fbc31139fabdb0090a325393974"
 
-  url "https://srf.baidu.com/?c=j&e=d&platform=mac"
+  url "https://imeres.baidu.com/mac/baiduinput_mac_v#{version.csv.first}_#{version.csv.second}.dmg"
   name "Baidu Input"
   name "安装百度输入法"
   homepage "https://srf.baidu.com/input/mac.html"
 
   livecheck do
-    url :url
-    strategy :extract_plist
+    url "https://srf.baidu.com/?c=j&e=d&platform=mac"
+    regex(/baiduinput[._-]mac[._-]v?(\d+(?:\.\d+)+)[._-](\w+)\.dmg/i)
+    strategy :header_match do |headers, regex|
+      headers["location"].scan(regex).map { |match| "#{match[0]},#{match[1]}" }
```

Comment:
```suggestion
      headers["location"]&.scan(regex)&.map { |match| "#{match[0]},#{match[1]}" }
```

I tend to use the safe navigation operator (`&.`) in `strategy` blocks when working with values that may not be present (e.g., if we receive an unexpected response).